src/term/mod.rs

```diff
@@ -873,8 +873,8 @@ impl SizeInfo {
         let line = Line(y.saturating_sub(self.padding_y as usize) / (self.cell_height as usize));
 
         Point {
-            line: min(line, self.lines() - 1),
-            col: min(col, self.cols() - 1)
+            line: min(line, self.lines()),
+            col: min(col, self.cols())
```

Comment:
I'm pretty certain that this breaks things, it's just not a code path which is commonly run.

Since `self.cols()` returns the number of columns, but the indexing starts at `0` and not `1`, this would introduce the classical issue you get with arrays when trying to index them with `array.length`.

Note that `self.lines()` and `self.cols()` should never be `0` in the first place, since that would mean there's not a single cell which exists. I'd just recommend going for `self.lines().saturating_sub(1)` for both lines and columns. That should solve the underflow **and** keep the logic accurate.